lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -27,16 +27,49 @@ class HttpClient
     #   :setting => value
     # }
 
-    def initialize(options={})
+    def initialize(options={})      
       @logger = options[:logger]
       # Again, in case we use DEFAULT_OPTIONS in the future, uncomment this.
       # @options = DEFAULT_OPTIONS.merge(options)
       @options = options
+
+      normalize_hosts!
+
+      @url_template = build_url_template
+
       @pool = build_pool(@options)
       # mutex to prevent requests and sniffing to access the
       # connection pool at the same time
     end
 
+    def build_url_template
+      {
+        :scheme => self.scheme,
+        :user => self.user,
+        :password => self.password,
+        :host => "URLTEMPLATE",
+        :port => self.port,
+        :path => self.path
+      }
+    end
+
+    # We support hostnames of the form 'myhostname' and 'myhostname:1234' where 1234 is the port
+    # The URI library does not parse this correctly, so here we normalize this so it is parsed as a hostname
+    # without a schema
+    HOSTNAME_PORT_REGEX=/\A(?<hostname>([A-Za-z0-9\.\-]+)|\[[0-9A-Fa-f\:]+\])(:(?<port>\d+))?\Z/   
+    def normalize_hosts!
+      @options[:hosts] = @options[:hosts].map do |host|
+        h_string = host.to_s
+
+        if HOSTNAME_PORT_REGEX.match(h_string)
+          # Making the scheme relative with // makes the host:port parse correctly
+          ::LogStash::Util::SafeURI.new("//#{h_string}")
+        else
+          host
```

Comment:
I think this whole block of code should probably be removed since the `SafeURI` constructor just does the right thing here. I believe the `host` variable is already a `SafeURI` at this point. Excellent catch!